Lib/idlelib/squeezer.py

```diff
@@ -212,6 +215,17 @@ def reload(cls):
             type="int", default=50,
         )
 
+        # Loading the font info requires a Tk root, and we don't want to
+        # rely on Tkinter's "default root", so each instance will load
+        # font info separately using its editor windows's Tk root.
+        new_instance_weakrefs = []
+        for inst_ref in cls._instance_weakrefs:
```

Comment:
For some reason I thought there was a way to have more than a single shell, but I was wrong.

We could make Squeezer a singleton and simplify this logic. I still think we'd need something of the sort, though, since the shell window can be closed and opened again.

I'll push an update soon.

Reply:
See the latest change which simplifies `Squeezer.reload()`.

Reply:
I want to sometime consider having a separate execution process for each file that is run, for the benefit of code completion.  That might or might not mean a separate Shell frame for each.  I would rather add code (as opposed to comments) needed for multiples shell only when needed, and would want to understand why.